scanner/trivy/scanner_test.go

```diff
@@ -0,0 +1,229 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trivy_test
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/aquasecurity/trivy/pkg/report"
+	"github.com/aquasecurity/trivy/pkg/types"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/rode/collector-image-scanner/scanner"
+	"github.com/rode/collector-image-scanner/scanner/trivy"
+	"github.com/rode/collector-image-scanner/scanner/trivy/trivyfakes"
+	"github.com/rode/rode/proto/v1alpha1fakes"
+	"github.com/rode/rode/protodeps/grafeas/proto/v1beta1/common_go_proto"
+	"github.com/rode/rode/protodeps/grafeas/proto/v1beta1/discovery_go_proto"
+	"github.com/rode/rode/protodeps/grafeas/proto/v1beta1/grafeas_go_proto"
+	"github.com/rode/rode/protodeps/grafeas/proto/v1beta1/package_go_proto"
+	"github.com/rode/rode/protodeps/grafeas/proto/v1beta1/vulnerability_go_proto"
+)
+
+var _ = Describe("TrivyScanner", func() {
+
+	var (
+		rode         *v1alpha1fakes.FakeRodeClient
+		trivyCommand *trivyfakes.FakeCommandWrapper
+		scanner      scanner.ImageScanner
+
+		expectedVersion string
+		versionError    error
+	)
+
+	BeforeEach(func() {
+		rode = &v1alpha1fakes.FakeRodeClient{}
+		trivyCommand = &trivyfakes.FakeCommandWrapper{}
+
+		expectedVersion = fake.Word()
+		versionError = nil
+
+		scanner = trivy.NewImageScanner(logger, rode, trivyCommand)
+	})
+
+	JustBeforeEach(func() {
+		trivyCommand.VersionReturns(&trivy.Version{Version: expectedVersion}, versionError)
+	})
+
+	Context("Init", func() {
+		var actualError error
+
+		JustBeforeEach(func() {
+			actualError = scanner.Init()
+		})
+
+		It("should check the Trivy version", func() {
+			Expect(trivyCommand.VersionCallCount()).To(Equal(1))
+		})
+
+		It("should not return an error", func() {
+			Expect(actualError).NotTo(HaveOccurred())
+		})
+
+		When("an error occurs checking the version", func() {
+			BeforeEach(func() {
+				versionError = errors.New("version error")
+			})
+
+			It("should return an error", func() {
+				Expect(actualError).To(HaveOccurred())
+			})
+		})
+	})
+
+	Context("ImageScan", func() {
+		var (
+			imageUri                 string
+			expectedOccurrencesCount int
+
+			expectedNoteName string
+			expectedNote     *grafeas_go_proto.Note
+			createNoteError  error
+
+			scanResults *trivy.ScanOutput
+			scanError   error
+		)
+
+		BeforeEach(func() {
+			expectedNoteName = fake.Word()
+			expectedNote = &grafeas_go_proto.Note{
+				Name: expectedNoteName,
+			}
+			createNoteError = nil
+
+			imageUri = fake.Word()
+			scanResults = &trivy.ScanOutput{
+				Report: &report.Report{},
+			}
+			scanError = nil
+
+			for i := 0; i < fake.Number(2, 5); i++ {
```

Comment:
this might not be doing what you expect it to.  `fake.Number(2, 5)` is actually executing with each iteration of the loop.